### PR TITLE
fix(database): Handle finding beetle data directory

### PR DESCRIPTION
**Pull request by flub**
This fixes very obscure errors when a beetle (formerly named iroh)
data directory is found.

- Error messages are improved to indicate what is failing.  Making use
  of anyhow's error chaining.

- The paths.bin file is used as a marker to decide if a database is
  present or needs to be created.

The end result is that this happily creates the new iroh database at
the same location of a previous beetle store, leaving the beetle files
in place.

### CODE REVIEW

**Review by dignifiedquire — APPROVED** [review]
looks reasonable to me, probably needs a look from the db chief @rklaehn

File: src/main.rs [hunk 1]
```diff
@@ -884,9 +891,9 @@ async fn get_interactive(get: GetInteractive, out: Option<PathBuf>) -> Result<()
                 // Rename temp file, to target name
                 let filepath2 = filepath.clone();
                 if let Some(parent) = filepath2.parent() {
-                    tokio::fs::create_dir_all(parent)
-                        .await
-                        .context("Unable to create directory {parent}")?;
+                    tokio::fs::create_dir_all(parent).await.with_context(|| {
```
Comment (rklaehn) [inline src/main.rs:894]: with_context is so that the formatting is lazy?

**Comment by rklaehn** [comment]
Looks reasonable.

**Review by rklaehn — DISMISSED** [review]

File: src/main.rs (hunk 1 above)
Comment (flub) [reply src/main.rs:894]: The previous version did not do any formatting, just literally printed `{parent}` in the output.

`.with_context` is if you need to do any anything that's not a static `Context` with a `Display`.  `format!` is just the most common version of this.

**Review by dignifiedquire — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (rklaehn) [reply src/main.rs]: oh, ok...

**Review by rklaehn — APPROVED** [review]

**Pull request merged**